### ISSUE TITLE
Sort Spellfixes.cpp into their proper files.

### ISSUE THREAD

**Issue by S7urp33** [S7urp33/ChronoEmu#12]
We need to break up individual class fixes into its proper locations under /spellsystem/ we can eventually do away with spellfixes.cpp if we create more in depth files like RaidSpellFixes.cpp, ItemsSpellFixes.cpp,etc. This allows better documentation. I will document where to get proper values for spellfixes and how to calculate formulas for spells that need to do certain damages based off tool tips to make this easier for new coming people..

**Comment by Swater96** [comment]
I agree, there are alot of spells with details in the core that should only be handled by the DB. 

Currently we're taking spells from DBC, modifying coefficients in the core, and also modifying coefficients in the DB. Finding out which is correct is going to be a PITA

**Comment by S7urp33** [comment]
Do you think it would be easier to just allow spell values to be changed via the database? This would allow a wider audience to fix them and not require them to look up values which we could store in the Wiki (I've been busy at work but I plan to set one up). In the long run as well it would save alot of lines of code. And instead of having to recompile the core each time to test a spell you could reload the individual tables possibly?

**Comment by Swater96** [comment]
Yeah there actually is a way to tune spell damage through the database, I remember when this core was live I used the database to slightly nerf Pyroblast. We can probably remove alot of core code by moving these things to DB

**Comment by S7urp33** [comment]
I noticed the things you talked about in the configs and code referencing WoTLK and TBC.. I'm going to try to clean that up tonight I've been doing a full data center overhaul at work because of a grant we got so everything has been out of wack the last 2 weeks.

**Comment by Swater96** [comment]
Of course no rush, I didn't want to remove them from the configs because they're referenced in many different files and have some opcodes included. Easier to track down when you know what you're looking for

**Comment by S7urp33** [comment]
This will be a major focus of mine this weekend, hopefully I can produce something for testing depending on my schedule I'm going to input the warden loading and checks tomorrow because I won't be home Thursday.

**Comment by S7urp33** [comment]
@Swater96 it looks like there is already a spellfixes table in the database and it loads into the core:

Line 1183 
File: ObjectMgr.cpp 
Function: LoadSpellFixes()

We could load more items into this such as is front facing, is a debuff, etc. Then start slowly clearing it out. We could also do damage base modifiers but it would require testing.. extensive because if any math is wrong a spell could do horrible amounts of damage.

**Comment by Swater96** [comment]
Agreed!... The thousands of hard-coded spells placed in 15 different files around the core makes me rage

**Comment by Swater96** [comment]
Currently i'm flirting with combining all the script projects into one, implementing the LUA engine, and consolidating all the battleground mess to the script libraries

**Comment by S7urp33** [comment]
@Swater96 Maybe combine them to compile with the core all in one like trinitycore does instead of individual dlls? The LUA engine I just didn't take the time to update that's my fault :( it was part of the project I just don't use LUA much.

**Comment by Swater96** [comment]
The problem with using scriptdev2 is that it is as bare bones as possible. Everything that can be handled by the database has been migrated there already. Once you integrate SD2 and EventAI then I feel like you're better off just forking mangos at that point.

**Comment by Swater96** [comment]
important to notice some of those tables are non-functional, especially spells112

**Comment by S7urp33** [comment]
When I start going through I'll check and remove non-functional tables I just listed anything with spells* prefix or regarding spells until further investigation.

**Comment by S7urp33** [comment]
I think that calculations should be a separate table spell_calculation or something like that so it doesn't get mixed in but there would have to be a field that you can choose between several methods like uses melee damage or increase with holy damage etc.
